safenode/src/domain/wallet/local_store.rs

```diff
@@ -496,6 +519,72 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn created_dbc_to_file_can_be_deposited() -> Result<()> {
+        // Bring in the necessary traits.
+        use super::{DepositWallet, SendWallet};
+
+        let sender_root_dir = create_temp_dir()?;
+        let sender_root_dir = sender_root_dir.path().to_path_buf();
+
+        let mut sender = LocalWallet::load_from(&sender_root_dir).await?;
+        let sender_dbc = create_genesis_dbc(&sender.key).expect("Genesis creation to succeed.");
+        sender.deposit(vec![sender_dbc]);
+
+        let send_amount = 100;
+
+        // Send to a new address.
+        let recipient_root_dir = create_temp_dir()?;
+        let recipient_root_dir = recipient_root_dir.path().to_path_buf();
+        let mut recipient = LocalWallet::load_from(&recipient_root_dir).await?;
+        let recipient_public_address = recipient.key.public_address();
+
+        let to = vec![(Token::from_nano(send_amount), recipient_public_address)];
+        let created_dbcs = sender.send(to, &MockSendClient).await?;
+        let dbc = created_dbcs[0].dbc.clone();
+        let dbc_id = dbc.id();
+        sender.store_created_dbc(dbc).await?;
+
+        let public_address_name = public_address_name(&recipient_public_address);
+        let public_address_dir = format!("public_address_{}", hex::encode(public_address_name));
+        let dbc_id_name = dbc_name(&dbc_id);
+        let dbc_id_file_name = format!("{}.dbc", hex::encode(dbc_id_name));
+
+        let created_dbcs_dir = sender_root_dir.join(WALLET_DIR_NAME).join("created_dbcs");
+        let created_dbc_file = created_dbcs_dir
+            .join(&public_address_dir)
+            .join(&dbc_id_file_name);
+
+        let received_dbc_dir = recipient_root_dir
+            .join(WALLET_DIR_NAME)
+            .join("received_dbcs")
+            .join(&public_address_dir);
+
+        tokio::fs::create_dir_all(&received_dbc_dir).await?;
+        let received_dbc_file = received_dbc_dir.join(&dbc_id_file_name);
+
+        // Move the created dbc to the recipient's received_dbcs dir.
+        tokio::fs::rename(created_dbc_file, &received_dbc_file).await?;
+
+        assert_eq!(0, recipient.wallet.balance().as_nano());
+
+        recipient.try_load_deposits().await?;
```

Comment:
I wasn't sure exactly, is this the method you're looking to cover with this test? You could consider putting that in the name of the test.

Also, I can see the setup might be a bit tedious, but it could be worth maybe having two or three DBCs, since I see it will attempt to load all of them. There was also a code path where there were no deposits. Is that worth having its own test case?

Reply:
Mm, yes, but not only. It's testing `store_created_dbc` and `try_load_deposits`. I can break them out into two tests, but the `try_load_deposits` need the first one to run anyway.

And do we name tests after the functionality/behaviour we want the system to have/display, or individual fn names?

Reply:
I renamed to `store_created_dbc_gives_file_that_try_load_deposits_can_use`. A bit long, but should be very clear at least :)